### ISSUE TITLE
Components Toolbar missing after adding block using keyboard only

### ISSUE THREAD

**Issue by jdelia** [WordPress/gutenberg#3782]
## Issue Overview
The component toolbar are missing after inserting a block using keyboard only and not moving the mouse.

## Steps to Reproduce (for bugs)

1. Add a block using the keyboard only (and not touching the mouse at all during the entire process)
2. After the block is inserted - using Shift-Tab or Tab does not reveal the component toolbar.
3. Moving to different blocks with the keyboard - you see that the component toolbar is missing for all of the blocks.
4. The momemt you touch and move the mouse - the component toolbar appears and works as expected. 

## Expected Behavior
The component toolbar should be available after inserting a block without using the mouse.

## Current Behavior
It does not do this.

## Possible Solution
This could be a bug that component toolbar is triggered only by mouse when inserting block.

## Screenshots / Video
https://cl.ly/3h0k332r1510

## Todos
- [ ] Tests
- [ ] Documentation

**Comment by afercia** [comment]
> After the block is inserted - using Shift-Tab or Tab does not reveal the component toolbar.

Actually, as soon as the block is inserted, the toolbar is displayed. As soon as you start typing, the toolbar disappears. See the posted video at 0:16. At that point, the above is true.

Using a mouse instead, the toolbar re-appears as soon as you move the mouse, even outside the block area. Selecting some text, makes the toolbar re-appear too.

However, when using a mouse, the toolbar re-appears when moving the mouse even if there's no selection. This helps, for example, for the alignment controls that don't require a selection.

Instead, to align a block with a keyboard (after you've typed something) you're stuck.  You have to first select something, then the toolbar re-appears, and then you can use Shift+Tab to go to the alignment controls.

I'd agree this different behavior is confusing and maybe there's still something to polish, re-think here. I'd also agree that, while typing, any control should disappear.

I guess the point is to clearly establish _when the toolbar should appear_. In keyboard navigation terms, what's the equivalent of moving the mouse? I guess the assumption here is that moving the mouse indicates the user's intention to do something else other than typing. 

Maybe the introduction of the "Navigation mode" and "Edit mode" currently being discussed could help. We could assume that, when in Edit mode, Shift-Tab or Tab indicates the user's intention to stop typing and do something else > toolbar re-appears.
Pressing Escape would switch to navigation mode: everything disappears.

/cc @jasmussen

**Comment by jasmussen** [comment]
CC: @mtias and @karmatosed which might not reply soon as I think we're all winding down for the holidays.

**Comment by jasmussen** [comment]
Seems like the issue could be mitigated a fair bit if we:

1. showed the block toolbar when using arrowkeys
2. showing the block toolbar when you shift tab

2 seems like it hinges on navigation mode (on my mockup todo list so we can revive that effort soon), and arrow-key navigation working sufficiently well for navigating between blocks. 

Would you agree with the above, Andrea?

**Comment by afercia** [comment]
I'd try 2 first.

1 could be confusing when navigating through blocks with up/down arrows?

**Comment by afercia** [comment]
Discussed during today's accessibility bug scrub and seems something has changed in the meantime. The toolbar appear on Tab or Shift-Tab. Instead, there's a different and potentially confusing behaviro related to the initial focus, for example:

Insert an image > initial focus is on the block container > the toolbar appears
Insert a Heading > initial focus is on the inner editable area > the toolbar doesn’t appear

This should be discussed in a separate issue though. Closing for now. Of course, can be reopened if we've missed something 🙂

**Comment by afercia** [comment]
Worth noting there's one more inconsistent behavior, to reproduce:

- use the "Add Block" button in the top left
- choose, for example, "Heading"
- a new Heading block gets added and the toolbar appears
- press Enter to create a new block
- use the Slash command and choose Heading
- a new Heading block gets added and the toolbar does not appear

**Comment by tofumatt** [comment]
This is not inconsistent behaviour to me: it is an intended effect of the slash-inserter which aims to be part of an expert user's uninterrupted writing flow. You can access the toolbar by reverse-Tab (Shift+Tab) when using the slash-inserter, but it is visually hidden to reduce visual noise when quickly adding a block using the keyboard.

I see how that last behaviour could be seen as inconsistent, but it is an indented effect. If you convert the block the menu appears, but the slash inserter should hide it to preserve writing flow.
